cpp/include/cudf/detail/utilities/device_operators.cuh

```diff
@@ -222,6 +222,15 @@ struct DeviceXor {
   }
 };
 
+/**
+ * @brief Operator for calculating Lead/Lag window function.
+ */
+struct DeviceLeadLag {
+  const size_type row_offset;
+
+  explicit CUDA_HOST_DEVICE_CALLABLE DeviceLeadLag(size_type offset_) : row_offset(offset_) {}
+};
```

Comment:
What is the purpose of this? It's not really an operator since it doesn't have an `operator()`.

Reply:
That's a fair point. I added this because `agg_op` is a parameter to `process_rolling`, but it isn't actually exercised there. 

I managed not to need this for `ROW_NUMBER`. I'll check if I can simply remove this.

Reply:
Nix that, I remember now.

This is the `ROW_NUMBER` device operator. It carries (to device code) the row-offset for `LEAD`/`LAG` calculation, i.e. how far forward/back to look for the answer.

I'm afraid we need this operator, even though it does not implement the function-call operator.